### PR TITLE
New settings option for a custom SSH host (#3763)

### PR DESCRIPTION
**Pull request by lunny**
cherry-pick from gogits/gogs#3763

### CODE REVIEW

File: conf/app.ini [hunk 1]
```diff
@@ -103,6 +103,8 @@ DISABLE_SSH = false
 START_SSH_SERVER = false
 ; Domain name to be exposed in clone URL
 SSH_DOMAIN = %(DOMAIN)s
+; Network interface builtin SSH server listens on
+SSH_LISTEN_HOST = 0.0.0.0
```
Comment (tboerger) [inline conf/app.ini]: Maybe empty by default? Than it should be also bound to ipv6?

Comment (lunny) [reply conf/app.ini]: OK. let it empty by default.

**Comment by tboerger** [comment]
Fixes https://github.com/go-gitea/gitea/issues/431

File: conf/app.ini (hunk 1 above)
Comment (lunny) [reply conf/app.ini]: Done.

**Comment by tboerger** [comment]
LGTM

**Comment by Bwko** [comment]
LGTM

**Pull request merged**